File: backend/app/models/discovery.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Optional


@dataclass(frozen=True)
class ScanJob:
    id: int
    created_at: str
    started_at: Optional[str]
    finished_at: Optional[str]
    created_by_user_id: Optional[int]
    status: str
    ip_start: str
    ip_end: str
    total_hosts: int
    processed_hosts: int
    found_hosts: int
    error_message: Optional[str]


@dataclass(frozen=True)
class ScanResult:
    id: int
    job_id: int
    ip_address: str
    port: int
    tcp_open: bool
    modbus_responding: bool
    response_time_ms: Optional[float]
    candidate_unit_ids: list[int]
    last_checked_at: str
    created_at: str

File: frontend/src/types/discovery.ts
export type ScanJobStatus = "pending" | "running" | "completed" | "failed" | "cancelled";

export interface ScanJob {
  id: number;
  created_at: string;
  started_at: string | null;
  finished_at: string | null;
  created_by_user_id: number | null;
  status: ScanJobStatus;
  ip_start: string;
  ip_end: string;
  total_hosts: number;
  processed_hosts: number;
  found_hosts: number;
  error_message: string | null;
  progress_percent: number;
}

export interface ScanResult {
  id: number;
  job_id: number;
  ip_address: string;
  port: number;
  tcp_open: boolean;
  modbus_responding: boolean;
  response_time_ms: number | null;
  candidate_unit_ids: number[];
  last_checked_at: string;
  created_at: string;
}

export interface ScanCreatePayload {
  ip_start: string;
  ip_end: string;
}
